### ISSUE TITLE
Tool doesn't exit once finished

### ISSUE THREAD

**Issue by VuokkoH** [DigitalGeographyLab/green-paths-2#10]
After a successful run, the tool doesn't exit properly. After `starting analysis pipeline` the tool outputs some `green_paths_2.src.timer - INFO` and the command line is left idle and needs to be restarted before starting a new run.

We have observed this issue when running gp2 on two different Windows 11 machines.

**Comment by VuokkoH** [comment]
I investigated this issue with the help of co-pilot (assumably fetching the information from discussion forums). Here is a summary of co-pilot's findings. I hope this helps if someone is planning to fix the issue. I have tested some different approaches. (Edit: I have fixed this now locally in an ugly way that forces the process to stop, but maybe there is a more professional way to fix this issue to the source code).

### Issue: JVM stays alive after CLI finishes (r5py / JPype shutdown issue)

**Description:**  
When running the GreenPaths 2 CLI via `invoke` (e.g.,
inv gp2 -a "-c green_paths_2/user/working_example_config.yaml all"
) the analysis completes, but the process does not exit. It appears the Java Virtual Machine (started by r5py/JPype) remains active even after all steps finish.

#### Observed behavior
- CLI output shows all analysis steps completed.
- The process stays idle and does not return to the shell.
- No error messages.

#### Possible cause
JPype’s `shutdownJVM()` (since ~0.7.5) waits for **non-daemon Java threads** before returning. If R5 or r5py leaves user threads alive, the JVM never shuts down gracefully, and the Python process hangs.

#### Potential directions
- Ensure any Python threads calling Java attach as daemon and detach when done:
  java.lang.Thread.attachAsDaemon()
  java.lang.Thread.detach()
- Add diagnostics to list Java threads before shutdown:
  java.lang.Thread.getAllStackTraces()
- Consider running the CLI in a **subprocess** so the JVM lifecycle is isolated; parent can enforce a timeout and kill if needed.
- Alternatively, provide an r5py API to cleanly stop internal Java threads before shutdown.
